### ISSUE TITLE
Using network retries causes client defined Idempotency Key to be over written

### ISSUE THREAD

**Issue by prettyboymp** [stripe/stripe-php#637]
When setting the `idempotency_key` option on a request, it gets overwritten if you are using network retries.  The issue is being caused by a check for an existing `Idempotency-Key` in the headers array by key, however, the `$headers` array was already converted to a non-keyed array in `ApiRequestor::_requestRaw()`.

Ref: https://github.com/stripe/stripe-php/blob/06c612bca4fcad33efa3cd54bc47942454eca031/lib/HttpClient/CurlClient.php#L208

Example:
```
\Stripe\Stripe::setMaxNetworkRetries( 2 );
\Stripe\Charge::create( $charge_parameters, [
  'api_key' => 'abcdefg',
  'idempotency_key' => '1234-5678-abcd-efg',
]);
```

The request getting sent to the API are getting a newly generated idempotency_key instead of the one manually set.

**Comment by ob-stripe** [comment]
Oops! Thanks for the report @prettyboymp, we'll work on a fix.

**Comment by ob-stripe** [comment]
Fixed in 6.34.2.
